exercises/transpose/examples/success-standard/src/Transpose.hs

```diff
@@ -0,0 +1,22 @@
+{-# LANGUAGE LambdaCase #-}
+
+module Transpose (transpose) where
+
+import Data.Maybe (isNothing)
+
+transpose :: [String] -> [String]
+transpose [] = []
+transpose rows =
+  map toString $ foldr 
+    (zipWith (:) . padRight . map Just) 
+    (replicate longestLength []) 
+    rows
+  where 
+    longestLength = maximum $ map length rows
+    padRight line = line ++ replicate (longestLength - length line) Nothing
+    toString line = concatMap (\case Nothing -> " "
+                                     Just c -> c:"") $ dropRight line
+    dropRight line = snd $ foldr 
```

Comment:
correct me if I'm wrong - is this https://hackage.haskell.org/package/base-4.10.1.0/docs/Data-List.html#v:dropWhileEnd ? If so, I think that should be used, for clarity.

Reply:
You are correct. I did not know about that function. Thanks for that!